### PR TITLE
Merge with upstream

### PR DESCRIPTION
**Pull request by jwmerrill**
Pull in all changes from mathquill/mathquill's master branch

### CODE REVIEW

File: src/commands/math/commands.js [hunk 1]
```diff
@@ -347,6 +363,28 @@ var SupSub = P(MathCommand, function(_, super_) {
       };
     }(this, 'sub sup'.split(' ')[i], 'sup sub'.split(' ')[i], 'down up'.split(' ')[i]));
   };
+  _.reflow = function() {
```
Comment (jwmerrill) [inline src/commands/math/commands.js:366]: This seems like a notable change. Landed in https://github.com/mathquill/mathquill/pull/779 as a fix for https://github.com/mathquill/mathquill/issues/763.

Comment (jwmerrill) [reply src/commands/math/commands.js:366]: Just opened https://github.com/mathquill/mathquill/issues/950 describing a problem with this behavior.

I'm not comfortable with the way this code seems to produce different appearance depending on the order you edit things in, even when those edits produce equivalent LaTeX. I'd prefer to back out this whole change before Desmos merges upstream changes.

Comment (jwmerrill) [reply src/commands/math/commands.js]: Note: I removed this code in 9ac1090.

File: src/commands/math.js [hunk 2]
```diff
@@ -26,6 +26,43 @@ var MathElement = P(Node, function(_, super_) {
     if (self[L].siblingCreated) self[L].siblingCreated(options, R);
     self.bubble(function (node) { node.reflow(); });
   };
+  // If the maxDepth option is set, make sure
+  // deeply nested content is truncated. Just return
+  // false if the cursor is already too deep.
+  _.prepareInsertionAt = function(cursor) {
+    var maxDepth = cursor.options.maxDepth;
+    if (maxDepth !== undefined) {
```
Comment (mikehaverstock) [inline src/commands/math.js:34]: My first question, without reading too much about what this new feature does, is should we be adding some of this "isTooDeep" logic to some of the code we've added?

**Review by mikehaverstock — APPROVED** [review]
Looks reasonable. My only question is about if this "isTooDeep" check needs added to code we introduced. I don't think this is a likely place, but for example, the shortcut to rendering basic numbers quickly might need a check. Or our formatting to add spaces between digits might need a check. Again, I think those places are almost certainly safe but maybe we introduced code I'm not thinking of that isn't as safe?

Also, after reviewing this I'm wondering how we plan to preserve blame history. Might be worth discussing strategies there rather than wiping all of mathquill's history if it's avoidable.

**Pull request merged**